### DISCUSSION TITLE
Support rounding of $__timeFilter macro

### DISCUSSION THREAD

**Discussion by dylan-asos** [grafana/grafana#57850] [Data Sources]
As per some of the points in #1412, a number of data sources now support query cache & would be useful to utilise them as much as possible and reduce load to source systems. 

In particular, the Azure Monitor data source supports a Logs query option via KQL, but with $__timeFilter producing a KQL statement at the second we're unlikely to benefit from caching if multiple users are viewing a dash, e.g. 

```
| where ['TimeGenerated'] >= datetime('2022-10-29T11:40:03Z') and ['TimeGenerated'] <= datetime('2022-10-29T12:10:03Z')
```

The ability to allow rounding to the nearest minute could help with this, e.g.

$___timeFilter(TimeGenerated, roundingUnit)
$___timeFilter(TimeGenerated, 1m)

```
| where ['TimeGenerated'] >= datetime('2022-10-29T11:40:00Z') and ['TimeGenerated'] <= datetime('2022-10-29T12:10:00Z')
```

Or (preferably) have it as a dashboard setting in the Time section, e.g.

![image](https://user-images.githubusercontent.com/16137664/198831038-b62d8ddd-95ee-471e-86f0-7643491ac22d.png)

This would allow applying the logic easily to all panels on dashboards without needing to change any queries - but obviously more work as all data sources would need their macros to support it.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
